src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs

```diff
@@ -0,0 +1,252 @@
+﻿
+using System;
+using System.Globalization;
+using System.Linq;
+using Microsoft.ServiceBus.Messaging;
+using Orleans.Providers.Streams.Common;
+using Orleans.Serialization;
+using Orleans.ServiceBus.Providers.Streams.EventHub;
+using Orleans.Streams;
+
+namespace OrleansServiceBusUtils.Providers.Streams.EventHub
+{
+    public struct CachedEventHubMessage
```

Comment:
I think it is very important it is a struct and all fields are primitives, right? Please add a comment about that, explaining why. We might even want to add a test to assert all that, so if someone changes it by mistake in the future, we will auto catch it.